scanner/src/main/kotlin/Scanner.kt

```diff
@@ -141,15 +141,46 @@ class Scanner(
             }
         }
 
+        val files = (projectResults.files + packageResults.files).distinctBy { it.provenance }.toSet()
+
+        val deduplicatedScanResults = (projectResults.scanResults + packageResults.scanResults)
+            .distinctBy { it.provenance to it.scanner }
+            .toSet()
+
+        val pathsByProvenance = files.associate { fileList ->
+            fileList.provenance to fileList.files.mapTo(mutableSetOf()) { it.path }
+        }
+
+        val checkedScanResults = if (scannerConfig.includeFilesWithoutFindings) {
+            deduplicatedScanResults.mapTo(mutableSetOf()) { scanResult ->
+                val allPaths = pathsByProvenance[scanResult.provenance].orEmpty()
+
+                val pathsWithFindings = scanResult.summary.licenseFindings.mapTo(mutableSetOf()) { it.location.path }
+                val pathsWithoutFindings = allPaths - pathsWithFindings
+
+                val findingsThatAreNone = pathsWithoutFindings.mapTo(mutableSetOf()) {
+                    LicenseFinding(SpdxConstants.NONE, TextLocation(it, TextLocation.UNKNOWN_LINE))
+                }
+
+                scanResult.copy(
+                    summary = scanResult.summary.copy(
+                        licenseFindings = scanResult.summary.licenseFindings + findingsThatAreNone
+                    )
+                )
+            }
+        } else {
+            deduplicatedScanResults
+        }
```

Comment:
Could we add a commit prior to this commit which factors out line 150-152 and lines 155 - 180 into a function which is responsible for adding these `LicenseFinding(SpdxConstants.NONE, TextLocation(it, TextLocation.UNKNOWN_LINE))`, so that the diff of this commit becomes simpler?

e.g. idea is to separate the de-duplication code from the adding of these NONE findings.

Reply:
See #10582